src/api_server/controllers/package_controller.ts

```diff
@@ -884,6 +887,53 @@ export async function package_byName_name_delete(req: Request, res: Response) {
  * 							PACKAGE_BYREGEX_REGEX_POST
  *
  */ ///////////////////////////////////////////////////////////////////////
-export function package_byRegEx_regex_post(req: Request, res: Response) {
-  res.status(200).send('This is wrong response btw');
+interface ReqRegEx {
+  RegEx: string;
+}
+export async function package_byRegEx_regex_post(req: Request, res: Response) {
+  const {RegEx} = req.body as ReqRegEx;
+  globalThis.logger?.debug(`Regex: ${RegEx}`);
+  try {
+    let result: any[] = [];
+
+    //Check if the regular expression is valid
+    new RegExp(RegEx);
+
+    //search for packages containing README files by regex
+    const packageByRegEx = await packages.findAll({
+      where: {
+        [Op.or]: [
+          {PackageName: {[Op.regexp]: RegEx}},
+          {ReadmeContent: {[Op.regexp]: RegEx}},
+        ],
+      },
+      attributes: ['VersionNumber', 'PackageName', 'PackageID'],
+    });
+
+    if (packageByRegEx) {
+      result = result.concat(packageByRegEx);
+    }
+    //globalThis.logger?.debug(`regex result: ${result[0].PackageName}`);
+    //globalThis.logger?.debug(`regex result length: ${result.length}`);
+
+    if (result.length !== 0) {
+      const updatedResult: PackageMetadata[] = [];
+      for (let i = 0; i < result.length; i++) {
+        const content: PackageMetadata = {
+          Name: result[i].PackageName,
+          Version: result[i].VersionNumber,
+          ID: result[i].PackageID.toString(),
+        };
+        globalThis.logger?.debug(`regex result: ${content}`);
+        updatedResult.push(content);
+      }
+      res.contentType('application/json').status(200).send(updatedResult);
+    } else {
+      res.status(404).send('No package found under this regex');
```

Comment:
I think we are not actually sending error messages on error, just log them
since 404 doesn't have content in the spec
still should be fine, but for consistency